src/lipyphilic/analysis/z_positions.py

```diff
@@ -197,6 +201,11 @@ def _prepare(self):
             fill_value=np.NaN,
         )
 
+        if self._return_midpoint:
+            self.memb_midpoint = np.full(
+                (self.n_frames, self.n_bins, self.n_bins),
+                fill_value=np.NaN)
```

Comment:
similarly here, the midpoint should be stored in `self.results.memb_midpoint` and a property `self.memb_midpoint` would return the midpoint stored in the `Results` class